k9mail/src/main/java/com/fsck/k9/fragment/AttachmentDownloadDialogFragment.java

```diff
@@ -0,0 +1,85 @@
+package com.fsck.k9.fragment;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.app.ProgressDialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+
+import com.fsck.k9.controller.MessagingController;
+import com.fsck.k9.controller.MessagingListener;
+import com.fsck.k9.controller.SimpleMessagingListener;
+
+public class AttachmentDownloadDialogFragment extends DialogFragment {
+
+    static ProgressDialog dialog;
+
+    static int progress = 0;
+
+    static MessagingListener messagingListener;
+
+    protected static final String ARG_SIZE = "size";
+    protected static final String ARG_MESSAGE = "message";
+
+    public static AttachmentDownloadDialogFragment newInstance(int size, String message) {
+        AttachmentDownloadDialogFragment attachmentDownloadDialogFragment = new AttachmentDownloadDialogFragment();
+
+        progress = 0;
+
+        Bundle args = new Bundle();
+        args.putInt(ARG_SIZE, size);
+        args.putString(ARG_MESSAGE, message);
+        attachmentDownloadDialogFragment.setArguments(args);
+
+        return attachmentDownloadDialogFragment;
+    }
+
+    public void updateDownloadProgress(int newProgress) {
+        progress = newProgress;
+        for (int i = progress; i <= newProgress; i++) {
+            dialog.setProgress(i);

```

Comment:
I added this just to make the transition smoother. Its not actually needed, will remove it shortly.